docs/quickstarts/java-quickstart.md

```diff
@@ -0,0 +1,192 @@
+
+You'll need a local copy of the example code to work through this quickstart. Download the example code from our Github repository (the following command clones the entire repository, but you just need the examples for this quickstart and other tutorials):
+
+```sh
+  $ # Clone the repository to get the example code:
+  $ git clone https://github.com/grpc/grpc-java
+  $ # Navigate to the "hello, world" Python example:
+  $ cd grpc/examples/python/helloworld
+  $ # Check out the latest release:
+  $ git checkout v1.0.0
+```
+
+## Run a gRPC application
+
+From the `examples` directory:
+
+1. Compile the client and server
+
+   ```sh
+   $ ./gradle installDist
+   ```
+
+2. Run the server
+
+   ```sh
+   $ ./build/install/examples/bin/hello-world-server
+   ```
+
+3. In another terminal, run the client
+
+   ```sh
+   $ ./build/install/examples/bin/hello-world-client
+   ```
+
+Congratulations! You've just run a client-server application with gRPC.
+
+## Update a gRPC service
+
+Now let's look at how to update the application with an extra method on the
+server for the client to call. Our gRPC service is defined using protocol
+buffers; you can find out lots more about how to define a service in a `.proto`
+file in [What is gRPC?]() and [gRPC Basics: Java][]. For now all you need
+to know is that both the server and the client "stub" have a `SayHello` RPC
+method that takes a `HelloRequest` parameter from the client and returns a
+`HelloResponse` from the server, and that this method is defined like this:
+
+
+```
+// The greeting service definition.
+service Greeter {
+  // Sends a greeting
+  rpc SayHello (HelloRequest) returns (HelloReply) {}
+}
+
+// The request message containing the user's name.
+message HelloRequest {
+  string name = 1;
+}
+
+// The response message containing the greetings
+message HelloReply {
+  string message = 1;
+}
+```
+Let's update this so that the `Greeter` service has two methods. Edit `examples/src/main/proto/helloworld.proto` and update it with a new `SayHelloAgain` method, with the same request and response types:
+
+```
+// The greeting service definition.
+service Greeter {
+  // Sends a greeting
+  rpc SayHello (HelloRequest) returns (HelloReply) {}
+  // Sends another greeting
+  rpc SayHelloAgain (HelloRequest) returns (HelloReply) {}
+}
+
+// The request message containing the user's name.
+message HelloRequest {
+  string name = 1;
+}
+
+// The response message containing the greetings
+message HelloReply {
+  string message = 1;
+}
+```
+
+(Don't forget to save the file!)
+
+## Update and run the application
+
+Normal compilation regenerates `GreeterGrpc.java`, which contains our generated client and server classes. It also regenerates classes for populating, serializing, and retrieving our request and response types.
+
+There's no need to worry about generating new server and client code, but we still need to implement and call the new method in the human-written parts of our example application.
+
```

Comment:
These two paragraphs are a bit unclear - how about:

When you recompile the example, as you'll do later, normal compilation will regenerate `GreeterGrpc.java`, which contains our generated gRPC client and server classes. This also regenerates classes for populating, serializing, and retrieving our request and response types.

However, you still need to implement and call the new method in the human-written parts of our example application.